sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3391,6 +3391,19 @@ object SQLConf {
       .intConf
       .createWithDefault(10000)
 
+  val ARROW_EXECUTION_MAX_RECORDS_PER_OUTPUT_BATCH =
+    buildConf("spark.sql.execution.arrow.maxRecordsPerOutputBatch")
```

Comment:
When would a user want to set this to a different value?

Reply:
Vectorized engines usually have a maximum batch size setting which prevents too big batch as input that possibly causes memory issue. For the downstream operators if they are custom vectorized operators, currently the Arrow output batch is sent to them as input. When user worries that the output batch might be too big for these operators, the user can set this config to limit the output batch size.